tests/kernel/mem_protect/stack_random/src/main.c

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
+#include <tc_util.h>
+#include <zephyr.h>
+
+#define STACKSIZE       2048
+#define THREAD_COUNT	64
+
+#if CONFIG_X86
+#define STACK_REG	"esp"
+#else
+#define STACK_REG	"sp"
+#endif
+
+#define VERBOSE		0
+
+void *last_sp = (void *)0xFFFFFFFF;
+unsigned int changed;
+
+void alternate_thread(void)
+{
+	register void *sp __asm__(STACK_REG);
```

Comment:
Does it work to just take the address of a stack variable? You won't have guarantees of where it is in the stack, but it should definitely move around as the stack is moved.

Reply:
I'll take a look